sklearn/ensemble/iforest.py

```diff
@@ -0,0 +1,245 @@
+            The input samples. Use ``dtype=np.float32`` for maximum
+            efficiency. Sparse matrices are also supported, use sparse
+            ``csc_matrix`` for maximum efficieny.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+
+        """
+        # ensure_2d=False because there are actually unit test checking we fail
+        # for 1d.
+        X = check_array(X, accept_sparse=['csc'], ensure_2d=False)
+        if issparse(X):
+            # Pre-sort indices to avoid that each individual tree of the
+            # ensemble sorts the indices.
+            X.sort_indices()
+
+        rnd = check_random_state(self.random_state)
+        y = rnd.uniform(size=X.shape[0])
+
+        # ensure that max_sample is in [1, n_samples]:
+        self.n_samples = X.shape[0]
+        if not (self.max_samples <= self.n_samples):
+            warn("max_samples is larger than the total number of samples"
+                 " n_samples. Corrected as max_samples=n_samples")
+            self.max_samples = self.n_samples
+        if not (0 < self.max_samples):
+            raise ValueError("max_samples has to be positive")
+
+        super(IsolationForest, self).fit(X, y, sample_weight=sample_weight)
+        return self
+
+    def _cost(self, n):
+        """ The average path length in a n samples iTree, which is equal to
+        the average path length of an unsuccessful BST search since the
+        latter has the same structure as an isolation tree.
+        """
+        if n <= 1:
+            return 1.
+        else:
+            harmonic_number = np.log(n) + 0.5772156649
+            return 2. * harmonic_number - 2. * (n - 1.) / n
+
+    def predict(self, X):
+        # code structure from ForestClassifier/predict_proba
+        """Predict anomaly score of X with the IsolationForest algorithm.
+
+        The anomaly score of an input sample is computed as
+        the mean anomaly scores of the trees in the forest.
+
+        The measure of normality of an observation given a tree is the depth
+        of the leaf containing this observation, which is equivalent to
+        the number of splitting required to isolate this point. In case of
+        several observations n_left in the leaf, the average lenght path of
+        a n_left samples isolation tree is added.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape (n_samples, n_features)
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        scores : array of shape (n_samples,)
+            The anomaly score of the input samples.
+            The lower, the more abnormal.
+        """
+        # Check data
+        X = check_array(X, dtype=DTYPE, accept_sparse="csr")
+        n_samples = X.shape[0]
```

Comment:
When you have rebased on current master, you will be able to add `check_is_fitted` here.